syscalls/src/cpi.rs

```diff
@@ -87,6 +87,41 @@ struct CallerAccount<'a> {
 }
 
 impl<'a> CallerAccount<'a> {
+    fn get_serialized_data(
+        memory_mapping: &MemoryMapping<'_>,
+        vm_addr: u64,
+        len: u64,
+        stricter_abi_and_runtime_constraints: bool,
+        account_data_direct_mapping: bool,
+    ) -> Result<&'a mut [u8], Error> {
+        if stricter_abi_and_runtime_constraints && account_data_direct_mapping {
+            Ok(&mut [])
```

Comment:
Maybe this could go with a `debug_assert!(false)`? You are not supposed to call this function when direct mapping is enabled.